GameEngine/Helpers/Helper.cs

```diff
@@ -0,0 +1,358 @@
+        /// <returns>Nearest carnivore to the animal.</returns>
+        private static Carnivore? CalculateMinDistanceToCarnivore(List<Carnivore> carnivoresAround, Animal animal)
+        {
+            double nearestCarnivoreDistance;
+            int counter = 0;
+            int nearestCarnivoreIndex = 0;
+            double minCarnivoreDistance = Double.MaxValue;
+
+            foreach (var carnivore in carnivoresAround)
+            {
+                PointsCoordinates pointsCoordinates = new PointsCoordinates
+                {
+                    FirstXPoint = animal.CoordinateX,
+                    SecondXPoint = carnivore.CoordinateX,
+                    FirstYPoint = animal.CoordinateY,
+                    SecondYPoint = carnivore.CoordinateY
+                };
+
+                nearestCarnivoreDistance = CalculateSquareDistanceByPythagoras(pointsCoordinates);
+
+                if (minCarnivoreDistance > nearestCarnivoreDistance)
+                {
+                    minCarnivoreDistance = nearestCarnivoreDistance;
+                    nearestCarnivoreIndex = counter;
+                }
+                counter++;
+            }
+
+            return carnivoresAround[nearestCarnivoreIndex];
+        }
+
+        /// <summary>
+        /// Checks the nearest hunter.
+        /// </summary>
+        /// <param name="nearestHunter">Nearest hunter.</param>
+        /// <param name="animal">Animal.</param>
+        /// <returns>Is hunter near.</returns>
+        public static bool IsHunterNear(IHunter nearestHunter, Animal animal) =>
+            Math.Abs(animal.CoordinateX - nearestHunter.CoordinateX) < 2 ||
+            Math.Abs(animal.CoordinateY - nearestHunter.CoordinateY) < 2;
+
+        /// <summary>
+        /// Calculates maximal distance to the carnivore by free cells.
+        /// </summary>
+        /// <param name="freeCells">Free cells.</param>
+        /// <param name="carnivore">Carnivore.</param>
+        /// <returns>Index of maximal distance cell.</returns>
+        public static int CalculateMaxDistanceFromCarnivoreByFreeCells(List<NewItemCoordinates> freeCells, Carnivore carnivore)
+        {
+            double distance;
+            int counter = 0;
+            int farthestFreeCellIndex = 0;
+            double maxDistance = 0;
+
+            foreach (var cell in freeCells)
+            {
+                PointsCoordinates pointsCoordinates = new PointsCoordinates
+                {
+                    FirstXPoint = cell.NewXCoordinate,
+                    SecondXPoint = carnivore.CoordinateX,
+                    FirstYPoint = cell.NewYCoordinate,
+                    SecondYPoint = carnivore.CoordinateY
+                };
+
+                distance = CalculateSquareDistanceByPythagoras(pointsCoordinates);
+
+                if (maxDistance < distance)
+                {
+                    maxDistance = distance;
+                    farthestFreeCellIndex = counter;
+                }
```

Comment:
After code block add new line and check this file, there is multiple places.

Reply:
Done.